### DISCUSSION TITLE
changing placeholder text "No file chosen" in the form input file label

### DISCUSSION THREAD

**Discussion by jeffhuang** [twbs/bootstrap#33880] [Q&A]
Is there a standard way of changing the placeholder text "No file chosen" in the form input file label? I looked through the docs and there weren't any examples of doing this. And the placeholder attribute doesn't seem to do it for file input.

This is for Bootstrap 5.0. Thanks!

**Comment by esitarz** [comment]
I believe this placeholder text is hardcoded via the browser and can't be changed.
Your "best" solution is probably to hide the input and/or input text, use a `<label for="formFile">` with your custom text and style as you see fit.

**Comment by ffoodd** [comment]
@esitarz is right. You may try to use v4 custom file inputs but it'll require some JavaScript to display selected files names.

**Comment by jeffhuang** [comment]
Alright, thank you!

**Comment by summer-cook** [comment]
Hey, [based on a comment on another issue](https://github.com/twbs/bootstrap/issues/32159#issuecomment-727717808) by @mdo , i understand that this has been changed in bootstrap 5. with BS5, I can now see the file names / number of files instead of "No file chosen" when i add the default version of this input. But, i need to use a custom onChange with the input like so (using react bootstrap):

```
<Form.Group controlId='attachments' className='mb-3'>
  <Form.Label>Attachments:</Form.Label>
  <Form.Control
      className='prevent-validation-styles'
      multiple
      type='file'
      onChange={handleAddFile}
      ref={fileRef}
  />
</Form.Group>
 ```
  with this onChange, i now only see "No files chosen". I guess the custom onChange is overriding the default onChange behavior that shows the file names/number of files.
  
  I was wondering, can anyone point me toward the source code for bootstrap 5 that makes the file names visible there? I looked through and couldn't seem to find where this is being handled. I would love to add this behavior back to my custom onChange if possible. thank you for the help!
